File: client/src/app/modules/quotation/services/notice.type.family.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { NoticeTypeFamily } from '../../quotation/model/NoticeTypeFamily';

@Injectable({
  providedIn: 'root'
})
export class NoticeTypeFamilyService extends AppRestService<NoticeTypeFamily>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getNoticeTypeFamilies() {
    return this.getListCached(new HttpParams(), "notice-type-families");
  }

  addOrUpdateNoticeTypeFamily(noticeTypeFamily: NoticeTypeFamily) {
    this.clearListCache(new HttpParams(), "notice-type-families");
    return this.addOrUpdate(new HttpParams(), "notice-type-family", noticeTypeFamily, "Enregistré", "Erreur lors de l'enregistrement");
  }
}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/NoticeTypeFamilyService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.NoticeTypeFamily;

public interface NoticeTypeFamilyService {
    public List<NoticeTypeFamily> getNoticeTypeFamilies();

    public NoticeTypeFamily getNoticeTypeFamily(Integer id);

    public NoticeTypeFamily addOrUpdateNoticeTypeFamily(NoticeTypeFamily noticeTypeFamily);

}
